olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -62,8 +61,11 @@ class DATASERVICE_READ_API VersionedLayerClient final {
    * @return A CancellationToken which can be used to cancel the ongoing
    * request.
    */
-  olp::client::CancellationToken GetData(DataRequest data_request,
-                                         DataResponseCallback callback);
+  olp::client::CancellationToken GetData(
+      DataRequest data_request,
+      std::function<
```

Comment:
naked like this is very bad, please use either CatalogClients aliases, or use approach proposed below.